spec/models/miq_report_result_spec.rb

```diff
@@ -1,14 +1,62 @@
 describe MiqReportResult do
-  context "persisting generated report results" do
-    before(:each) do
-      EvmSpecHelper.local_miq_server
+  before :each do
+    EvmSpecHelper.local_miq_server
+
+    @user1 = FactoryGirl.create(:user_with_group)
+  end
+
+  context "report result created by User 1 with current group 1" do
+    before :each do
+      @report_1 = FactoryGirl.create(:miq_report)
+      group_1 = FactoryGirl.create(:miq_group)
+      group_2 = FactoryGirl.create(:miq_group)
+      @user1.miq_groups << group_1
+      @report_result1 = FactoryGirl.create(:miq_report_result, :miq_report_id => @report_1.id, :miq_group => group_1)
+      @report_result2 = FactoryGirl.create(:miq_report_result, :miq_report_id => @report_1.id, :miq_group => group_1)
+      @report_result_nil_report_id = FactoryGirl.create(:miq_report_result)
+
+      @report_2 = FactoryGirl.create(:miq_report)
+      @report_result3 = FactoryGirl.create(:miq_report_result, :miq_report_id => @report_2.id, :miq_group => group_2)
+      User.current_user = @user1
+    end
+
+    describe ".with_report" do
+      it "returns report all results without nil report_id" do
+        report_result = MiqReportResult.with_report
+        expect(report_result).to match_array([@report_result1, @report_result2, @report_result3])
+        expect(report_result).not_to include(@report_result_nil_report_id)
+      end
+
+      it "returns only requested report results" do
+        report_result = MiqReportResult.with_report(@report_result1.miq_report_id)
+        expect(report_result).to match_array([@report_result1, @report_result2])
+        expect(report_result).not_to match_array([@report_result3, @report_result_nil_report_id])
+      end
+    end
 
-      @user = FactoryGirl.create(:user_with_group)
+    describe ".with_current_user_groups" do
+      it "returns report results by generated by user 1, non-admin user logged" do
```

Comment:
Ugh, spelling.  `returns report results generated for the current non-admin user`